server/routes.go

```diff
@@ -156,7 +157,10 @@ func PullModelHandler(c *gin.Context) {
 			Password: req.Password,
 		}
 
-		if err := PullModel(req.Name, regOpts, fn); err != nil {
+		ctx, cancel := context.WithCancel(c.Request.Context())
+		defer cancel()
```

Comment:
Is this necessary? `c` itself implements `context.Context` and should already be cancellable

Reply:
It doesnt cancel the context on disconnect when I dont have this

Reply:
In that case, you should use `context.WithCancel(c)` since `gin.Context` implements the `Context` interface

Reply:
This does not work either. Looks like the gin context matches the interface but its not intended to be used this way. Its more HTTP context specific.
https://stackoverflow.com/questions/57270597/transform-gin-context-to-context-context

Reply:
Hmm that's surprising and unfortunate